### PR TITLE
New modules added: issues #200 and #310

### LINKED ISSUE #200: new module: gatk4/genotypegvcf
I think it would be good to have module for [gatk4](https://bioconda.github.io/recipes/gatk4/README.html)

### PR DESCRIPTION
**Pull request by santiagorevale**
## PR checklist

Closes #200 and #310.

Added new modules: GATK4 genotypegvcfs (#200) and indexfeaturefile (#310).

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by santiagorevale** [comment]
Hi @drpatelh, we are good to go, here. All parties are happy with the implementation. I also implemented your changes, thank you so much! Cheers!

File: tests/modules/gatk4/genotypegvcfs/main.nf [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+    GATK4_GENOTYPEGVCFS ( input, fasta, fastaIndex, fastaDict, dbsnp, dbsnpIndex, intervalsBed )
+}
+
+
+//
+// Download the "test_genomicsdb" folder and all its contents to use in a test
+//
+def getTestGenomicsDbData() {
+    test_genomicsdb = file("test_genomicsdb")
```
[the remark this answers is not in the document]
Comment (santiagorevale) [reply tests/modules/gatk4/genotypegvcfs/main.nf]: this has been properly sorted out using the test datasets

**Review by drpatelh — APPROVED** [review]

**Comment by GCJMackenzie** [comment]
Hi @santiagorevale module is merged now! Will need to update the gatk version to 4.2.3 at a later date and update the md5's for the tests when that happens. I also found a slightly better way to add empty optional inputs. But I am happy to implement these changes later if you are happy with me doing so. :)

**Comment by santiagorevale** [comment]
Sure, GC, no problem. Could you show me the way to handle the empties?

**Comment by GCJMackenzie** [comment]
@santiagorevale An example is here: https://github.com/GCJMackenzie/modules/blob/add_gatk_somatic_tumour_calling_subworkflow/subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/main.nf#:~:text=ch_stats%20%3D%20%20%20%20%20%20%20%20%20MUTECT2.out,out.segmentation.collect()

I used .add([]) to put the empty placeholder in. It's important to note I add the empty onto the end of the stats list before it is combined with the other channels. This is because add() only works with lists and not channels and combine() transforms the list into a channel, so the empty must be added first.

**Comment by santiagorevale** [comment]
where would you implement this?

**Pull request merged**